### PR TITLE
betterbird: work around ICU-22132

### PR DESCRIPTION
**Pull request by charmoniumQ**
To trigger the buggy behavior, go to Menu > Events and Tasks > Calendar, and the calendar UI element will not render at all.

Go to Menu > Tools > Developer Tools > Error Console, and see that there is a JS exception "TypeError: zone is null".

## Description of changes

Apply the same fix to Betterbird that #243179 applied to Thunderbird

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/betterbird/default.nix [hunk 1]
```diff
@@ -110,6 +112,22 @@ in ((buildMozillaMach {
...
+  # [this issue][1] which is ultimately due to a [bug in Unicode ICU][2].
+  # Therefore, Betterbird should apply the same patch that Thunderbird does at
+  # present (see ../thunderbird/packages.nix, or [permalink][3]).
+  #
+  # [1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1843007
+  # [2]: https://unicode-org.atlassian.net/browse/ICU-22132
+  # [3]: https://github.com/NixOS/nixpkgs/blob/2c7f3c0fb7c08a0814627611d9d7d45ab6d75335/pkgs/applications/networking/mailreaders/thunderbird/packages.nix#L82
+  icu = icu.overrideAttrs (attrs: {
+    patches = attrs.patches ++ [(fetchpatch2 {
+      url = "https://hg.mozilla.org/mozilla-central/raw-file/fb8582f80c558000436922fb37572adcd4efeafc/intl/icu-patches/bug-1790071-ICU-22132-standardize-vtzone-output.diff";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/betterbird/default.nix:126]: Can't we passthru thunderbirds ICU and use that?

File: pkgs/applications/networking/mailreaders/betterbird/default.nix [hunk 2]
```diff
@@ -110,6 +112,22 @@ in ((buildMozillaMach {
   webrtcSupport = false;
 
   pgoSupport = false; # console.warn: feeds: "downloadFeed: network connection unavailable"
+
+  # Betterbird is broken the same way Thunderbird is regarding VTZONE parsing in
+  # [this issue][1] which is ultimately due to a [bug in Unicode ICU][2].
+  # Therefore, Betterbird should apply the same patch that Thunderbird does at
+  # present (see ../thunderbird/packages.nix, or [permalink][3]).
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/betterbird/default.nix:119]: The comment is way to long. We can drop this

File: pkgs/applications/networking/mailreaders/betterbird/default.nix [hunk 3]
```diff
@@ -110,6 +112,22 @@ in ((buildMozillaMach {
...
+  #
+  # [1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1843007
+  # [2]: https://unicode-org.atlassian.net/browse/ICU-22132
+  # [3]: https://github.com/NixOS/nixpkgs/blob/2c7f3c0fb7c08a0814627611d9d7d45ab6d75335/pkgs/applications/networking/mailreaders/thunderbird/packages.nix#L82
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/betterbird/default.nix:123]: And that line. It is the same git repo which shouldn't be to hard to find again.

File: pkgs/applications/networking/mailreaders/betterbird/default.nix (hunk 1 above)
Comment (charmoniumQ) [reply pkgs/applications/networking/mailreaders/betterbird/default.nix:126]: I'm afraid I don't kwow what that means.

**Comment by ibrokemypie** [comment]
Hey, whats the reason this pr was closed? I still have the same issue with tasks/calendar not working and have been using [charmoniumQ:patch-betterbird](https://github.com/charmoniumQ/nixpkgs/tree/patch-betterbird) in my flake ever since.

**Comment by charmoniumQ** [comment]
Closed because I couldn't figure out how what "add the patched binary to passthru and re-use it here" means. It would appear I broke something.

**Comment by SuperSandro2000** [comment]
I did a minimal version of this PR in #344071

**Pull request closed without merging**